src/api/app/helpers/webui/repository_helper.rb

```diff
@@ -0,0 +1,41 @@
+module Webui::RepositoryHelper
+  def effective_flag(flags, repository, architecture)
+    dig_flag(:effective_flag_for, flags, repository, architecture)
+  end
+
+  def default_flag(flags, repository, architecture)
+    dig_flag(:default_flag_for, flags, repository, architecture)
+  end
+
+  def flag_set_by_user?(flags, repository, architecture)
+    flags[:object].dig(repository, architecture).present?
+  end
+
+  def html_id_for_flag(flag_type, repository, architecture)
+    # repository and architecture can be nil
+    valid_xml_id("flag-#{flag_type}-#{repository}-#{architecture}")
+  end
+
+  # TODO: This should be private
+  def dig_flag(callback_function, flags, repository, architecture)
+    flag = send(callback_function, flags[:object], repository, architecture)
+    flag ||= send(callback_function, flags[:project], repository, architecture) if flags[:project]
```

Comment:
The callback function is always specified by us or is it possible for the user?

Reply:
We can't make functions in modules private, but it's just a helper function for default_flag and effective_flag